frontend/apps/reader/modules/readerpagemap.lua

```diff
@@ -348,9 +348,8 @@ function ReaderPageMap:addToMainMenu(menu_items)
                     self.use_page_labels = not self.use_page_labels
                     self.ui.doc_settings:saveSetting("pagemap_use_page_labels", self.use_page_labels)
                     -- Reset a few stuff that may use page labels
-                    self.ui.toc:resetToc()
-                    self.ui.view.footer:onUpdateFooter()
-                    self.ui.annotation:updatePageNumbers(true)
+                    local Event = require("ui/event")
+                    UIManager:broadcastEvent(Event:new("UseReferencePageUpdate"))
```

Comment:
Please make the event name accurate codewise:
- ending with Update**d** (we signal something did happen, we are not requesting for something to be done, giving the order to update something)
- "ReferencePageUpdate" is bad: we are not updating one singular page :) but we use Reference Page**s**
- the variable name and setting names are called `use_page_labels` (might not be the nicest choice for a name, but it's here everywhere, "Reference page numbers" is just the user facing name.

So I guess the event name should be: `UsePageLabelsUpdated`.